### PR TITLE
Ability to load from Gists

### LINKED ISSUE #3: Ability to retrieve project.json from a gist?
We need some ability to try to retrieve project.json (and maybe save it) to make it easy to share projects.

### PR DESCRIPTION
**Pull request by kitsonk**
This PR provides the ability to load projects from Gists, where the `project.json` is stored in a Gist.

The README contains information on how to utilise the new features and there is now a `examples/gist.html` and `examples/gist.ts` which demonstrate integrating the features in a UI.

Fixes #3

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,8 +56,8 @@ Typical usage would be something like this:
 import project from '@dojo/web-editor/project';
 
 (async () => {
-	await project.load('some.project.json');
-	console.log('Loaded!');
+    await project.load('some.project.json');
```
Comment (dylans) [inline README.md]: Looks like the readme was changed from tabs to spaces throughout?

File: README.md [hunk 2]
```diff
@@ -118,13 +118,49 @@ import project from '@dojo/web-editor/project';
...
+### routing
+
+This module wraps `@dojo/routing` in a way that works well for the `web-editor`.
+
+The `web-editor` allows integration with GitHub's Gists, allowing storage of the `project.json` in a Gist which can then be used to load
+the project into the editor.
+
+#### setPath()
+
+Set the path on the current page.  Intended to be used, when a project is loaded to update the path to reflect the Gist's ID.
```
Comment (dylans) [inline README.md]: "Intended to be used, when a project is loaded to update the path to reflect the Gist's ID." -> When a project is loaded, this is used to update the path to reflect the Gist's ID.

File: src/examples/gist.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+async function loadProjectButtonClick() {
+	projectSelect.setAttribute('disabled', 'disabled');
+	loadProjectButton.setAttribute('disabled', 'disabled');
+	await loadProject(projectSelect.value);
+	titleH2.textContent = projectSelect.selectedOptions[0].text;
+	setPath(gistIDMap.get(projectSelect.value!)!);
+}
+
+/**
+ * Use the username input to retrieve all the public gists for a user, loking for those that contain a
```
Comment (dylans) [inline src/examples/gist.ts]: loking -> looking

File: README.md (hunk 1 above)
Comment (kitsonk) [inline README.md]: That was intentional... tabs are just causing problems in the README from spacing, when it gets rendered it looks totally unnatural and unreadable... 😭

**Pull request merged**